packages/lit-html/src/directives/async-append.ts

```diff
@@ -125,3 +125,9 @@ class AsyncAppendDirective extends AsyncDirective {
  *     value. Useful for generating templates for each item in the iterable.
  */
 export const asyncAppend = directive(AsyncAppendDirective);
+
+/**
+ * Non-callable type of the directive class. Necessary for when a function or
+ * method returns the return type of the above directive.
+ */
+export type {AsyncAppendDirective as AsyncAppendDirectiveType};
```

Comment:
For confusion and DX:

For someone not too familiar with lit they may see both class and directive implemented and try to instantiate the class and apply it.
Additionally, if we explort the class of `AsyncAppendDirective` and `asyncAppend` it'll likely pollute autocomplete imports and confuse users. I did call them AsyncAppendDirectiveClass, but peter wasn't a fan (and type was also more correct for this case)

Reply:
I think it's ok to export the class here. It's more straightforward and we don't have to come up with new names. The class object is available on the directive results, so we're not leaking any reference we weren't before.

Reply:
Exports cost bytes though 😇

Reply:
Yes, though not much, only a bit per-directive module, and it facilitates subclassing if we cared about that. We already export `UnsafeHTML` and might get requests to export others like `RepeatDirective`.

I would say if we export the types only that we should:
1. Make the class names consistent: always end in `Directive`
2. Export the type with the plain name, ex `RepeatDirective, and name the class something like `RepeatDirectiveImpl`
3. Don't add an extra type export for classes already exports. So in unsafe-html.ts we'd just keep exporting the class.

This way if we ever decide to export a class later we just remove the type export and rename the class and don't end up with two exports a type and a class.

Reply:
So the change would look like:

```ts
export type UntilDirective = UntilDirectiveImpl;
class UntilDirectiveImpl extends AsyncDirective { ... }
export const until = directive(UntilDirectiveImpl);
```

Reply:
I still don't understand why we should be exporting something we don't expect or really want users to use

Reply:
Why is unsafe-html exporting a class? Do users need it?

Reply:
It's so it can be subclassed by `UnsafeSVG`. Like I said, I could see someone wanting to subclass directives like repeat. We don't have to export them now, but if we do in the future it'd be nice if it was a minimal change and we didn't end up with two exports that represent the type.

`XImpl` is renamed by the minifier internally. That name doesn't matter. Let's use the good, simple name for the exported name.

Reply:
on the first part, i see it's used for unsafe svg